File: java-language/src/test/java/com/github/daggerok/javalang/MainTest.java
package com.github.daggerok.javalang;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class MainTest {

  @Test
  void test() {
    var args = new String[0];
    assertNotNull(args);
    Main.main(args);
  }
}

File: scala-language/src/test/scala/com/github/daggerok/scalalang/MainTest.scala
package com.github.daggerok.scalalang

import org.junit.runner.RunWith
import org.scalatest.{FunSuite, Matchers}
import org.scalatest.junit.JUnitRunner

@RunWith(classOf[JUnitRunner])
class MainTest extends FunSuite with Matchers {
  test("should Hello, Scala!") {
    val hello = Hello()
    hello should not be None
    hello.message should not be None
    hello.greet() should equal("Hello, Scala!")

    val args = Array[String]()
    args should have size 0
    Main.main(args)
  }
}

